code/go/0chain.net/smartcontract/provider/provider.go

```diff
@@ -0,0 +1,70 @@
+package provider
+
+import (
+	"fmt"
+
+	"0chain.net/core/common"
+)
+
+//go:generate msgp -io=false -tests=false -v
+
+// swagger:enum Status
+type Status int
+
+const (
+	Active Status = iota
+	Inactive
+	ShutDown
+	Killed
+	NonExistent
+)
+
+var statusString = []string{"active", "inactive", "shut_down", "killed", "non_existent"}
+
+func (p Status) String() string {
+	return statusString[p]
+}
+
+// swagger:model StatusInfo
+type StatusInfo struct {
+	Status Status `json:"status"`
+	Reason string `json:"reason"`
+}
+
+type ProviderI interface {
+	Status(now common.Timestamp) Status
+	HealthCheck(now common.Timestamp) error
+	ShutDown() error
+	Kill() error
+}
+
+type Provider struct {
+	LastHealthCheck common.Timestamp `json:"last_health_check"`
+	IsShutDown      bool             `json:"is_shut_down"`
+	IsKilled        bool             `json:"is_killed"`
+}
+
+func (p *Provider) Status(now, healthCheckPeriod common.Timestamp) (Status, string) {
```

Comment:
The string value is not the string version of the status, but the reason why the status is not active. For example it could be inactive because
* health check failed
* killed or shout down
* write price is too which due to change in global max write price
* read price is too which due to change to global max read price
* capacity is too low due to change in min blobber capacity
(https://github.com/0chain/0chain/blob/provider_changes/code/go/0chain.net/smartcontract/storagesc/models.go#L386)